admins/pageflow/entry.rb

```diff
@@ -50,28 +59,48 @@ module Pageflow
-    filter :with_publication_state, as: :select, collection: -> { collection_for_entry_publication_states }
-
-    sidebar :folders, :only => :index do
-      text_node(link_to(I18n.t('pageflow.admin.entries.add_folder'), new_admin_folder_path, :class => 'new'))
-      grouped_folder_list(Folder.includes(:account).accessible_by(Ability.new(current_user), :read),
-                          :class => authorized?(:manage, Folder) ? 'editable' : nil,
-                          :active_id => params[:folder_id],
-                          :grouped_by_accounts => authorized?(:read, Account))
+    filter :with_publication_state,
+           as: :select,
+           collection: -> { collection_for_entry_publication_states }
+
+    sidebar :folders, only: :index do
+      text_node(link_to(I18n.t('pageflow.admin.entries.add_folder'),
+                        new_admin_folder_path,
+                        class: 'new'))
+      grouped_folder_list(Folder.accessible_by(Ability.new(current_user), :read),
+                          active_id: params[:folder_id],
+                          grouped_by_accounts: true)
     end
 
     form do |f|
       f.inputs do
-        f.input :title, :hint => I18n.t('pageflow.admin.entries.title_hint')
-        if authorized?(:read, Account)
-          f.input :account, :include_blank => false
+        f.input :title, hint: I18n.t('pageflow.admin.entries.title_hint')
 
-          unless f.object.new_record?
-            f.input :theming, :include_blank => false
-          end
+        eligible_accounts = AccountPolicy::Scope.new(current_user, Pageflow::Account)
+                            .entry_movable.load
+        if authorized?(:update_account_on, resource)
+          f.input :account,
+                  collection: eligible_accounts,
+                  include_blank: false,
+                  input_html: {onchange: jquery_for_disabling_on_change}
+        end
+
+        unless f.object.new_record?
+          eligible_themings = ThemingPolicy::Scope.new(current_user, Pageflow::Theming)
+                              .themings_allowed_for(resource.account).load
+        end
+
+        if authorized?(:update_theming_on, resource) && !f.object.new_record?
+          f.input :theming, collection: eligible_themings, include_blank: false
         end
-        if authorized?(:manage, Folder)
-          f.input :folder, :collection => collection_for_folders(f.object.account), :include_blank => true
+
+        if authorized?(:configure_folder_for, resource) || controller.action_name == :create
+          resource_folder_id = resource.folder ? resource.folder.id : nil
+          f.input(:folder,
+                  collection: collection_for_folders(resource.account,
+                                                     resource_folder_id),
+                  include_blank: true) unless collection_for_folders(resource.account,
+                                                                     resource_folder_id).empty?
```

Comment:
The local variable can be avoided here if we pass `folder` directly to the helper. Maybe instead store the result of `collection_for_folders` in a local to prevent hitting the database twice and simplify the `unless` condition.